File: DFS,BFS/2606_바이러스.py
import sys
input = sys.stdin.readline

node = int(input())
graph = [[0] for _ in range(node+1)]
visited = [0 for _ in range(node+1)]
C = int(input())

for i in range(C):#making graph
    n1, n2 = map(int, input().split())
    graph[n1].append(n2)
    graph[n2].append(n1)

cnt=0  
def DFS(graph, visited, node):
    visited[node] = 1
    global cnt
    for i in graph[node]:
        if visited[i]==0:
            DFS(graph, visited, i)
            cnt+=1

DFS(graph, visited, 1)
'''
for i in range(1, node+1):
    if visited[i] == 0:
        DFS(graph, visited, i)
'''
print(cnt-1)

File: DFS,BFS/2606_바이러스.cpp
#include<iostream>
#include<vector>
using namespace std;
int cnt=0;
void DFS(vector<int> *graph, bool *visited, int node){
	visited[node]=true;
	for(int i=0;i<graph[node].size();i++){
		int next = graph[node][i];
		
		if(visited[next]==false){
			DFS(graph, visited, next);
			cnt+=1;
		}	
	}
}
int main(){
	int node, C;
	cin>>node>>C;
	vector<int> graph[node+1];
	
	for(int i=0;i<C;i++){
		int n1, n2;
		cin>>n1>>n2;
		graph[n1].push_back(n2);
		graph[n2].push_back(n1);
	}
	
	bool visited[node+1]={false};
	DFS(graph,visited,1);
	
	cout<<cnt;
	
}
